website/common/script/ops/unlock.js

```diff
@@ -1,18 +1,69 @@
 import get from 'lodash/get';
-import each from 'lodash/each';
-import pick from 'lodash/pick';
 import setWith from 'lodash/setWith';
 import i18n from '../i18n';
-import splitWhitespace from '../libs/splitWhitespace';
-import {
-  NotAuthorized,
-  BadRequest,
-} from '../libs/errors';
+import { NotAuthorized, BadRequest } from '../libs/errors';
 
 import { removeItemByPath } from './pinnedGearUtils';
 import getItemInfo from '../libs/getItemInfo';
 import content from '../content/index';
 
+const incentiveBackgrounds = ['blue', 'green', 'red', 'purple', 'yellow'];
+
+function determineCost (isBackground, isFullSet) {
+  if (isBackground) {
+    return isFullSet ? 3.75 : 1.75;
+  }
+  return isFullSet ? 1.25 : 0.5;
+}
+
+function isGear (path) {
+  return path.includes('gear.');
+}
+
+function alreadyOwns (user, path) {
+  return get(user, isGear(path) ? path : `purchased.${path}`);
```

Comment:
I think this is correct for non gear items but for gear, you can actually lose an unlocked gear item in which case `user.items.gear.owned[key]` will be `false`  but you still own it (can buy it back with gold and not gems) so I would check for the key missing (user has never owned the gear):

```
return isGear(path) 
    ? get(user, path) !== undefined;
    : get(user, `purchased.${path}`) === true; // so that it's converted to a boolean
```

Reply:
Ah yeah, good point. However the conversion to boolean is not necessary right? The field is boolean in the database, so what you'd essentially be doing is `true === true`.

Reply:
BTW how then do you buy back the lost gear item? Is that somewhere in the buy endpoints?

Reply:
And won't the singles flow break down? Because I think it'll try to set the lost item as preferred. Will look in to working around that.

Reply:
> Ah yeah, good point. However the conversion to boolean is not necessary right? The field is boolean in the database, so what you'd essentially be doing is true === true.

yes but being a Mixed field when it does not exist it won't be false but simply undefined, so if you don't have the item `get(user, `purchased.${path}`)` will be `undefined` not `false`

> BTW how then do you buy back the lost gear item? Is that somewhere in the buy endpoints?

> And won't the singles flow break down? Because I think it'll try to set the lost item as preferred. Will look in to working around that.
> Edit: Guess I'll have to introduce a difference between alreadyOwns and alreadyUnlocked


let me check

Reply:
Ah, of course, the preferences field doesn't have any item field in the schema, that's why it doesn't matter. Will leave it as is then and only fix the bug.